### DISCUSSION TITLE
Is there a way to programatically add the missing keys on the JSON?

### DISCUSSION THREAD

**Discussion by aguilaair** [vinissimus/next-translate#581] [Q&A] [answered]
Hey,

Firstly thank you so much for this package! We're migrating from a CRA application to Next.js, in the process switching to `next-translate`. My question is if there is a way to run a command to extract all keys and add them to their respective json?

Right now we're using `i18next-parser` but it is not ideal for this package. I've tried searching around but have found nothing so any help would be appreciated.

Thanks!

**Comment by aralroca** [comment, marked as answer]
@aguilaair there isn't right now. Maybe we can build one, It can be useful

**Reply by aguilaair** [reply]
That would be awesome. Thanks @aralroca!

**Comment by camelaissani** [comment]
I have written a script which extracts the i18n keys from the source code and it merges them in the en locale file: https://github.com/microrealestate/frontend/blob/master/scripts/extractlocalizedstrings.js
It is basic but it might be a good starting point.

**Comment by wuifdesign** [comment]
i created a package for this purpose, hope it works fine.

https://www.npmjs.com/package/next-translate-scanner

if you like it, you may add it to your readme
